### DISCUSSION TITLE
Live Update:  Bundle does not contain a signature.

### DISCUSSION THREAD

**Discussion by Santosl2** [capawesome-team/capacitor-plugins#250] [Q&A]
Hello

i'm using Next JS 14.2.2 and attempting to implement Live Update in My app, but i'm encountering a 'Bundle does not contain a signature.' error. I have already read and implemented Code Signing, as described in [this guide](https://capawesome.io/cloud/live-updates/advanced/code-signing/).

![image](https://github.com/user-attachments/assets/81da3df6-43d5-44de-b066-fe5a661500ee)

## My code implementation
![image](https://github.com/user-attachments/assets/83299903-ee20-472d-ad78-41151f44d6e3)

## My capacitor.config.json
![image](https://github.com/user-attachments/assets/18882dd2-32f2-4230-a380-eb711212780c)

### When i run `LiveUpdate.getBundles()` a empty array is returned
![image](https://github.com/user-attachments/assets/1814295f-9908-4b35-85de-72d6af5b9d67)

**Comment by robingenz** [comment]
You need to create a signed bundle using your private key: 

```
npx capawesome apps:bundles:create --private-key keypair.pem
```

Please make sure that you are using the latest Capawesome CLI version:

```
npm install -g @capawesome/cli@latest
```

**Reply by Santosl2** [reply]
Nice dude! The issue was because i hadn't installed the capawesome CLI, but now i'm encountering a different problem:

When i run: `getBundle()` function its returns 'undefined'.

My app still isn't updating, see:

![image](https://github.com/user-attachments/assets/af417814-7254-4edd-8321-d1ce44453d7d)

**Reply by robingenz** [reply]
`getBundle()` only returns the currently active bundle. If you call `sync()` you just download and set the **next** bundle. You have to call `reload()` or restart the app to use finally apply the new bundle. After that, if you call `getBundle()` you should receive the new bundle id.

```ts
/**
 * Call this method to download and set the latest bundle for the app.
 */
const sync = async () => {
    const result = await LiveUpdate.sync();
    if (result.nextBundleId) {
        await LiveUpdate.reload();
    }
};
```
